src/runtime/HalideBuffer.h

```diff
@@ -76,10 +76,15 @@ struct AllocationHeader {
 
 /** A similar struct for managing device allocations. */
 struct DeviceRefCount {
+    enum struct Ownership : int {
+        Allocated,
```

Comment:
Should probably add comments here detailing exact semantics of each.... it may seem self evident but given the complexity of failure modes we've been seeing it's probably better to over-specify.